UniversalAlgebra/GaloisConnections/Basic.lagda

```diff
@@ -39,46 +42,57 @@ module _ {α α₁ α₂ : Level}(A : Poset α α₁ α₂)
   _≤A_ = _≤_ A
   _≤B_ = _≤_ B
 
- record Galois : Type  (α ⊔ α₂ ⊔ β ⊔ β₂) where
+ record Galois : Type (lsuc ℓ)  where
   field
    F : Carrier A → Carrier B
    G : Carrier B → Carrier A
-   F⊣G : ∀ a b → (F a) ≤B b → a ≤A (G b)
-   F⊢G : ∀ a b → a ≤A (G b) → (F a) ≤B b
+   GF≥id : ∀ a →  a ≤A G (F a)
+   FG≥id : ∀ b →  b ≤B F (G b)
+
+ record Residuation : Type (lsuc ℓ)  where
+  field
+   f : Carrier A → Carrier B
+   g : Carrier B → Carrier A
+   gf≥id : ∀ a → a ≤A g (f a)
+   fg≤id : ∀ b → f (g b) ≤B b
 
 
-private variable ℓ ρ : Level
 
+module _ {ℓ : Level}{𝒜 ℬ : Type ℓ} where
 
-module _ {𝒜 ℬ : Type ℓ} where
+ infix 10 _⃗_ _⃖_
 
  -- For A ⊆ 𝒜, define A ⃗ R = {b : b ∈ ℬ,  ∀ a ∈ A → R a b }
- _⃗_ : (A : Pred 𝒜 (ℓ ⊔ ρ)) (R : REL 𝒜 ℬ ρ) → Pred ℬ (ℓ  ⊔ ρ)
+ _⃗_ : (A : Pred 𝒜 ℓ) (R : REL 𝒜 ℬ ℓ) → Pred ℬ ℓ
  A ⃗ R = λ b → (a : 𝒜) → a ∈ A → R a b
```

Comment:
Would it make sense to write this as
```
A ⃗ R = λ b → (a : 𝒜) → a ∈ A → b ∈ R a
```
?  *Or* to use `_~_` (or some other infix notation, maybe even `_R_`) ?

Reply:
Yes, I tried to define this in a slicker way.  Indeed, the condition is more simply stated as follows:

```
 _⃗_ : (A : Pred 𝒜 ℓ) (R : REL 𝒜 ℬ ℓ) → Pred ℬ ℓ
 A ⃗ R = λ b → A ⊆ (λ a → R a b)
```
However, when I changed to this presentation, it made the proof of the following harder:

```
←→≥id : {A : Pred 𝒜 ℓ} {R : REL 𝒜 ℬ ℓ} → A ⊆ R ⃖ (A ⃗ R)
```

Reply:
Thanks for all your excellent comments and observations, Jacques!  Extremely helpful!!

Reply:
Glad you feel that way. I'd really want to dig right in, I just don't quite have the time right now.